pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -555,6 +555,15 @@ public static class Server {
     // Use 10 seconds by default so high volume stream are able to catch up.
     // This is also the default in the case a user misconfigures this by setting to <= 0.
     public static final int DEFAULT_STARTUP_REALTIME_MIN_FRESHNESS_MS = 10000;
+    // The timeouts above determine how long servers will poll their status before giving up.
+    // This configuration determines what we do when we give up. By default, we will mark the
+    // server as healthy and start the query server. If this is set to true, we instead throw
+    // an exception and exit the server. This is useful if you want to ensure that the server
+    // is always fully ready before accepting queries. But note that this can cause the server
+    // to never be healthy if there is some reason that it can never reach a GOOD status.
+    public static final String CONFIG_OF_EXIT_SERVER_ON_INCOMPLETE_STARTUP =
+        "pinot.server.starter.exitServerOnStartupStatusFailure";
```

Comment:
Move it next to `CONFIG_OF_STARTUP_ENABLE_SERVICE_STATUS_CHECK`
```suggestion
    public static final String CONFIG_OF_STARTUP_EXIT_ON_SERVICE_STATUS_CHECK_FAILURE =
        "pinot.server.startup.exitOnServiceStatusCheckFailure";
```

Reply:
done

Reply:
I'm also suggesting changing the key to make it more specific. See the suggested change